File: Selenium/Activities/Activity10.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC


with webdriver.Firefox() as ff:

    ff.get("https://v1.training-support.net/selenium/dynamic-controls")
    print("title is : ", ff.title)
    wait = WebDriverWait(ff, 10)
    toggleButton = ff.find_element(By.ID, "toggleCheckbox")
    checkbox = ff.find_element(By.XPATH, "//div/input[@type='checkbox']")

    print("Check box is visible : ", checkbox.is_displayed())

    toggleButton.click()
    wait.until(EC.invisibility_of_element(checkbox))
    print("Check box is visible : ", checkbox.is_displayed())
    ff.quit()

File: Selenium/Activities/Activity10.java
package activities;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Activity10 {

	public static void main(String[] args) {
		WebDriver driver = new FirefoxDriver();
		driver.get("https://v1.training-support.net/selenium/dynamic-controls");

		System.out.println("The Title of the page is  : " + driver.getTitle());
		WebElement toggleButton = driver.findElement(By.id("toggleCheckbox"));
		WebElement checkbox = driver.findElement(By.xpath("//div/input[@type='checkbox']"));

		System.out.println("Check box is visibile : " + checkbox.isDisplayed());
		toggleButton.click();

		System.out.println("Check box is visibile : " + checkbox.isDisplayed());

		driver.quit();

	}

}
